src/hyperlight_host/src/hypervisor/hyperv_windows.rs

```diff
@@ -561,51 +553,44 @@ impl Hypervisor for HypervWindowsDriver {
-                    access_info,
-                ) {
-                    Some(access_info) => access_info,
-                    None => HyperlightExit::Mmio(gpa),
+                match access_info {
+                    MemoryRegionFlags::READ => HyperlightExit::MmioRead(gpa),
+                    MemoryRegionFlags::WRITE => HyperlightExit::MmioWrite(gpa),
+                    _ => HyperlightExit::Unknown("Unknown memory access type".to_string()),
                 }
             }
-            //  WHvRunVpExitReasonCanceled
-            //  Execution was cancelled by the host.
-            WHV_RUN_VP_EXIT_REASON(8193i32) => {
-                debug!("HyperV Cancelled Details :\n {:#?}", &self);
-
-                HyperlightExit::Cancelled()
-            }
+            // Execution was cancelled by the host.
+            WHvRunVpExitReasonCanceled => HyperlightExit::Cancelled(),
             #[cfg(gdb)]
-            WHV_RUN_VP_EXIT_REASON(4098i32) => {
-                // Get information about the exception that triggered the exit
+            WHvRunVpExitReasonException => {
                 let exception = unsafe { exit_context.Anonymous.VpException };
 
-                match self.get_stop_reason(exception) {
-                    Ok(reason) => HyperlightExit::Debug(reason),
-                    Err(e) => {
-                        log_then_return!("Error getting stop reason: {}", e);
+                // Get the DR6 register to see which breakpoint was hit
+                let dr6 = {
+                    let names = [WHvX64RegisterDr6];
+                    let mut out: [Align16<WHV_REGISTER_VALUE>; 1] = unsafe { std::mem::zeroed() };
+                    unsafe {
+                        WHvGetVirtualProcessorRegisters(
+                            self.partition,
+                            0,
+                            names.as_ptr(),
+                            1,
+                            out.as_mut_ptr() as *mut WHV_REGISTER_VALUE,
+                        )?;
                     }
+                    unsafe { out[0].0.Reg64 }
+                };
+
+                HyperlightExit::Debug {
+                    dr6,
+                    exception: exception.ExceptionType as u32,
                 }
             }
-            WHV_RUN_VP_EXIT_REASON(_) => {
-                debug!(
-                    "HyperV Unexpected Exit Details :#nReason {:#?}\n {:#?}",
-                    exit_context.ExitReason, &self
-                );
-                match self.get_exit_details(exit_context.ExitReason) {
-                    Ok(error) => HyperlightExit::Unknown(error),
-                    Err(e) => HyperlightExit::Unknown(format!("Error getting exit details: {}", e)),
-                }
-            }
+            WHV_RUN_VP_EXIT_REASON(_) => HyperlightExit::Unknown(format!(
+                "Unknown exit reason '{}'",
+                exit_context.ExitReason.0
```

Comment:
Why are we directly accessing a member here vs calling a function previously, what changed?

Reply:
OK, so we no longer capture registers in the log? Again I think that was there because it was used in the past to try and narrow down where issues were (registers such as RIP are/were useful to try and determine what\where errors were happening) I suspect that this isnt as important now that we have crashdump capability

Reply:
If we want this, lets add it back to all in hyperlight_vm so it applies to all vms, not just windows. But if it's only used for non-live debugging, I say lets get rid of it. If you need extra info while debugging offline, you can add whatever you want